news-explorer/src/components/Main/Main.js

```diff
@@ -7,49 +7,38 @@ import NewsCardList from '../NewsCardList/NewsCardList';
 import SavedNewsHeader from '../SavedNewsHeader/SavedNewsHeader';
 import About from '../About/About';
 import { AuthContext } from '../../contexts/AuthContext';
-import { UserContext } from '../../contexts/UserContext';
 import { HomeProvider } from '../../contexts/HomeContext';
 import { Route, Routes } from 'react-router-dom';
 
-
-
-function Main({ openAuthPopup, newsData, isOpen, saveOrDelArticle }) {
-  const [isLoading, setIsLoading] = useState(false);
-  const [hasResults, setHasResults] = useState(false);
+function Main({ openAuthPopup, newsData, isOpen, sendSearchQuery, saveOrDelArticle, setInfoOpen, newsResults, preLoader, setPreloader }) {
   const { isLoggedIn } = useContext(AuthContext);
   const articleList = Object.values(newsData);
-  const { userData } = useContext(UserContext);
-
-  function getArticleList() { 
-    setHasResults(false);
-    setIsLoading(false);
-  }
-
   return (
     <>
       <main className="main">
         <HomeProvider>
-        <Routes>
-          <Route path='/'
-            element={<>
-              <div className="main__header-search-container">
-                <Header openAuthPopup={openAuthPopup} isOpen={isOpen} />
-                <SearchForm getArticleList={getArticleList} />
-              </div>
-              <Preloader isLoading={isLoading} hasResults={hasResults} />
-              <NewsCardList articleList={articleList} saveOrDelArticle={saveOrDelArticle}/>
-              <About />
-            </>} />
-          {isLoggedIn && (
-            <Route path='/saved-news'
+          <Routes>
+            <Route path='/'
               element={<>
-                <Header openAuthPopup={openAuthPopup} isOpen={isOpen} />
-                <SavedNewsHeader articleList={articleList} username={userData.name} />
+                <div className="main__header-search-container">
+                  <Header openAuthPopup={openAuthPopup} isOpen={isOpen} />
+                  <SearchForm sendSearchQuery={sendSearchQuery} setInfoOpen={setInfoOpen} setPreLoader={setPreloader} />
+                </div>
+                <Preloader preLoader={preLoader} />
+                {(!preLoader.isLoading && newsResults.data.length !== 0) &&
+                  <NewsCardList newsResults={newsResults} saveOrDelArticle={saveOrDelArticle} preLoader={preLoader} />}
+                <About />
               </>} />
-          )}
-        </Routes>
+              {isLoggedIn && (
+              <Route path='/saved-news'

```

Comment:
* I can visit `Saved artices` without loggin in https://disk.yandex.ru/i/ncewJi1-rih53g . But it should be protected


You need to have `ProtectedRoute` for that